docker_e2e.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
```

Comment:
Mind moving this file into a subdirectory `test`?

Reply:
Please add `set -eo pipefail` such that test failures bubble up as proper exit codes.